### PR TITLE
fix(userspace): variable / function shadowing

### LINKED ISSUE #3122: Fix variable / function shadowing

### PR DESCRIPTION
**Pull request by sgaist**
**What type of PR is this?**

This PR refactors the code around variable and function shadowing. While the code is working, shadowing makes reasoning about it harder.

As a drive by refactor, variables names have been improved around the code changed.

Reported by cppcheck.

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> /kind release

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area engine

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3122

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by incertum** [comment]
I like these changes for enhanced clarity. Curious @jasondellaluce what are your thoughts since you worked the most on these areas?

File: userspace/falco/app/actions/init_inspectors.cpp [hunk 1]
```diff
@@ -197,10 +197,9 @@ falco::app::run_result falco::app::actions::init_inspectors(falco::app::state& s
 			const auto& sources = src_info->inspector->event_sources();
 			if (sources.size() == 0 || sources.size() > 2 || sources[0] != falco_common::syscall_source)
 			{
-				std::string err;
-				for (const auto &s : sources)
+				for (const auto &source : sources)
```
Comment (Andreagit97) [inline userspace/falco/app/actions/init_inspectors.cpp:200]:
```suggestion
                                err.clear();
				for (const auto &source : sources)
```

Comment (Andreagit97) [reply userspace/falco/app/actions/init_inspectors.cpp:200]: just to be sure

**Review by Andreagit97 — COMMENTED** [review]
thank you for this! left just one nit

File: userspace/falco/app/actions/init_inspectors.cpp (hunk 1 above)
Comment (jasondellaluce) [reply userspace/falco/app/actions/init_inspectors.cpp:200]: +1

**Review by Andreagit97 — APPROVED** [review]
/approve

**Pull request merged**